### ISSUE TITLE
"ilinebr" on every link broke my entire project

### ISSUE THREAD

**Issue by aotto1968** [doxygen/doxygen#7948]
**Describe the bug**
I get the following error messages (multiple)

1. End of list marker found without any preceding list items ( I have no clue where)
2. "ilinebr" attached on !! many !! links so that near everything is broken 

```
warning: End of list marker found without any preceding list items
warning: unable to resolve reference to 'tcl_ObjectObject\ilinebr' for \ref command
warning: unable to resolve reference to 'tcl_ErrorObject\ilinebr' for \ref command
```

"linebr" is part of your SW

```
src/commentscan.l:DOCNL     "\n"|"\\ilinebr"
src/commentscan.l:<Comment>{B}*"\\ilinebr"{B}*            { // preserve spacing around \\ilinebr
src/commentscan.l:<Comment>(\n|\\ilinebr)({B}*(\n|\\ilinebr))+ { // at least one blank line (or blank line command)
src/commentscan.l:                                              else if (strcmp(yytext+i,"\\ilinebr")==0) addOutput(yyscanner,"\\ilinebr"),i+=8;
src/commentscan.l:                                              else if (strcmp(yytext+i,"\\ilinebr")==0) addOutput(yyscanner,"\\ilinebr"),i+=8;
src/commentscan.l:<SectionTitle>[^\n@\\]*/"\\ilinebr"     { // end of section title
src/commentscan.l:    else if (c=='r' && i>=7 && qstrncmp("\\ilinebr",s.data()+i-7,8)==0) // special line break marker
src/doctokenizer.l:  int nl2 = text.findRev("\\ilinebr");
src/doctokenizer.l:<St_Para>{BLANK}*(\n|"\\ilinebr"){LISTITEM}     { /* list item on next line */
src/doctokenizer.l:<St_Para>{BLANK}*(\n|"\\ilinebr"){MLISTITEM}     { /* list item on next line */
src/doctokenizer.l:<St_Para>{BLANK}*(\n|"\\ilinebr"){OLISTITEM}     { /* list item on next line */
src/doctokenizer.l:<St_Para>{BLANK}*(\n|"\\ilinebr"){ENDLIST}      { /* end list on next line */
src/doctokenizer.l:<St_Para>"\\ilinebr"   {
src/doctokenizer.l:<St_CodeOpt>"\\ilinebr" |
src/doctokenizer.l:<St_HtmlOnlyOption>"\\ilinebr" {
src/doctokenizer.l:<St_PlantUMLOpt>{BLANK}*{FILEMASK}{BLANK}*/"\\ilinebr" { // case 5: plain file name specified without title or attributes
src/doctokenizer.l:<St_PlantUMLOpt>"\\ilinebr" |
src/doctokenizer.l:<St_Title>"\\ilinebr"  {
src/doctokenizer.l:<St_TitleN>(\n|"\\ilinebr") { /* new line => end of title */
src/doctokenizer.l:<St_TitleQ>(\n|"\\ilinebr") { /* new line => end of title */
src/doctokenizer.l:<St_TitleV,St_TitleA>(\n|"\\ilinebr") {
src/doctokenizer.l:<St_Cite>(\n|"\\ilinebr")  { // new line
src/doctokenizer.l:<St_Ref>(\n|"\\ilinebr") { // new line
src/doctokenizer.l:<St_Ref2>"\""|\n|"\\ilinebr" { /* " or \n => end of title */
src/doctokenizer.l:<St_SkipTitle>(\n|"\\ilinebr") { return 0; }
src/doctokenizer.l:<St_SecSkip>(\n|"\\ilinebr")
src/doctokenizer.l:<St_Sections>(\n|"\\ilinebr")
src/doctokenizer.l:                         if (g_secTitle.right(8)=="\\ilinebr")
src/doctokenizer.l:<St_Snippet>(\n|"\\ilinebr")  {
src/doctokenizer.l:<*>"\\ilinebr"         {
src/doxygen.cpp:        newValue+="\\ilinebr ";
src/doxygen.cpp:      newValue+="\\ilinebr ";
src/util.cpp:    else if (c=='\\' && qstrncmp(p,"\\ilinebr",8)==0) i+=8,li=i,docLine++,p+=8;
src/util.cpp:    else if (c=='r' && b>=7 && qstrncmp(p-7,"\\ilinebr",8)==0) bi=b-7,b-=8,p-=8;
src/markdown.cpp:// test if the next characters in data represent a new line (which can be character \n or string \ilinebr).
src/markdown.cpp:  if (data[0] == '\\' && qstrncmp(data+1,"ilinebr",7)==0) return 8;
src/markdown.cpp:      m_out.addStr("> " + cellText + "\\ilinebr </" + cellTag + ">");
build/generated_src/commentscan.cpp:{ // preserve spacing around \\ilinebr
build/generated_src/commentscan.cpp:                                              else if (strcmp(yytext+i,"\\ilinebr")==0) addOutput(yyscanner,"\\ilinebr"),i+=8;
build/generated_src/commentscan.cpp:                                              else if (strcmp(yytext+i,"\\ilinebr")==0) addOutput(yyscanner,"\\ilinebr"),i+=8;
build/generated_src/commentscan.cpp:    else if (c=='r' && i>=7 && qstrncmp("\\ilinebr",s.data()+i-7,8)==0) // special line break marker
build/generated_src/doctokenizer.cpp:  int nl2 = text.findRev("\\ilinebr");
build/generated_src/doctokenizer.cpp:                         if (g_secTitle.right(8)=="\\ilinebr")
```

**Expected behavior**
no warning -> just do the job

**To Reproduce**
impossible - it is my entire project :-(

**Version**
1.8.19 (84084bebe0544650513dbcd75e71a50b2119360d)

**Comment by aotto1968** [comment]
one source of the bug is an ref in a "markdown table" - the "tcl_ObjectObject" in the markdown-table should expand to "MqObjectC" like the text under the table  (MqObjectC do some stuff) -
>> Problem: markdown table

![doxygen_Bug3](https://user-images.githubusercontent.com/27886664/89692375-cdde7f80-d90b-11ea-9baa-b90a7800741f.png)

**Comment by albert-github** [comment]
I see, at the moment, the following problems:
1. warning: End of list marker found without any preceding list items
  This typically happens when there is a dot (`.`) at the beginning of the line.
2. warning: unable to resolve reference to 'tcl_ObjectObject\ilinebr' for \ref command
  This problem is caused by the not handling of a `\ilinebr` case in `doctokinizer.l`. The `\ilinebr` result from a `^^` in an `ALIASES`.

As you can understand it will be a bit hard to find the cause of the problem (other than some general remarks) without a small example. Is your code propriety code or can you share it (still a small example would be best)? 

Regarding the first problem based on the remark above and the line number as mentioned with the with the warning it should be possible to pinpoint the problem (note a guess is that it might also come from an `ALIASES` where there is `^^.`).

Regarding the second problem, this looks like a forgotten case. Also here probably you will be able to pinpoint the problem a bit bu means of the line / file as given with the warning, based on this you probably van create a s small example. At least giving the `ALIASES` might already help here.

In both cases adjusting the `INPUT` tag only to one relevant file for each problem might help considerably (of course some other warnings might pop up but those can be ignored during the investigations).

**Comment by aotto1968** [comment]
thanks for the help… the cause was on missing "}" in a very large project … this created an extremely flood of error-messages without any relationship to the inital missing "}" -

-> it is an problem if the error-messages are not related to the initial problem

The problem was at the "system_primitive_types" item:

```
 ...
  <dt>  SYSTEM    \anchor \DocNS{idx_system}
    <dd>
      \RNS{system}                      "Philosophy"      ,
      \RNS{system_primitive_types      "PrimitiveTypes"  ,
      \RNS{system_managed_object}       "ManagedObject"   ,
      \RNS{system_storage_management}   "StorageManagement"
  <dt>  CLASS     \anchor \DocNS{idx_class}
    <dd>
...
```

**Comment by albert-github** [comment]
So when I understand it correctly the problem is solved and was "usage"

Didn't you get a message like:
```
Reached end of file while still inside a (nested) comment.  Nesting level 1 (probable line reference: 4)
```

Still not 100% clear but giving an indication of where the problem occurred.

**Comment by aotto1968** [comment]
yes - I saw this message but together with ~1000++ other messages - and I don't understand that this message point to a missing "}" - I for example search for a problem with "\if -> \endif" etc … you don't see a missing "}" in a large block of code…

The problem of debugging was:  sometimes it works (with an other setup) and sometimes not - sometimes doxygen complains about aliases which are defined but not found… etc. -> part of the output was generated - part not -> it was a total "random" error.

the core problem was - I thing - that doxygen run out of internal memory without "}" and than start to overwrite existing memory…

valgrind/sanitze etc does NOT complain -> this seems to be a sign that already allocated memory was "overwritten" inside doxygen…

**Comment by albert-github** [comment]
Doxygen tries to match the open and close `{` / `}` and similar, and here it was missing a `}` and running to the end of the code wile doxygen thinks that also the comment close is part of, here, the argument of the `\RNS`.

**Comment by albert-github** [comment]
Even though the problem is a usage problem, the remark:
> The problem of debugging was: sometimes it works (with an other setup) and sometimes not - sometimes doxygen complains about aliases which are defined but not found… etc. -> part of the output was generated - part not -> it was a total "random" error.

made me think a bit ...

I tried a similar example with `///` and `/**` comments and 
- with `///` I didn't get a message but the comment was silently omitted
- with the `/**` comment we got the warnings and also some output.

aa.h
```
/// \file

/// the fie
///
///   \RNS{system_primitive_types      "PrimitiveTypes"  ,<br>
///
void fie(void);
```
bb.h
```
/** \file

    the fie

      \RNS{system_primitive_types      "PrimitiveTypes"  ,<br>
*/
void fie1(void);
```

Doxyfile:
```
ALIASES += "RNS{1}=RNS \1"
QUIET=YES
```

The warnings:
```
.../bb.h:9: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
.../bb.h:7: warning: Member fie1(void) (function) of file bb.h is not documented.
.../bb.h:5: warning: Found unknown command '\RNS'
```

I would have expected that also in the case of `///` there would have been a message.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/5045143/example.tar.gz)

**Comment by aotto1968** [comment]
I use the same code with multiple targets for ONE target I get the

    warning: Reached end of file while still inside a (nested) comment
  
and for the other target not - I don't get the message:

    warning: Found unknown command '\RNS' 

but a lot of other like:

    warning: Found unknown command '\function'
   
The "function" alias has nothing todo with the "RNS" alias -> probably a scaling problem and a memory overrun.

I use the /** ... */ syntax in this file.

**Comment by albert-github** [comment]
Regarding the missing `}` I've just pushed a proposed patch, pull request #10686 which emits a warning regarding the problem.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
